controllers/dspipeline_controller.go

```diff
@@ -374,10 +375,11 @@ func (r *DSPAReconciler) setStatusAsUnsupported(conditionType string, err error,
 	setStatus(condition)
 }
 
-func (r *DSPAReconciler) setStatus(ctx context.Context, resourceName string, conditionType string,
+func (r *DSPAReconciler) setStatus(ctx context.Context, resourceName string, conditionType string, statusMessage string,
 	dspa *dspav1.DataSciencePipelinesApplication, setStatus func(metav1.Condition),
 	log logr.Logger) {
 	condition, err := r.evaluateCondition(ctx, dspa, resourceName, conditionType)
+	condition.Message = statusMessage
```

Comment:
I'm not sure if overwriting the `Message` here is the right call. The reason I'm wondering that is that https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/dspipeline_controller.go#L406 has a number of instances in which it is setting the `Message` field itself. Thoughts @hbelmiro ?

Reply:
I actually proposed replacing
```go
r.setStatus(ctx, params.MlmdProxyDefaultResourceName, config.MLMDProxyReady, statusMessage, dspa,
				dspaStatus.SetMLMDProxyStatus, log)
```

with

```go
			condition, err := r.evaluateCondition(ctx, dspa, params.MlmdProxyDefaultResourceName, config.MLMDProxyReady)
			condition.Message = status
			if err != nil {
				r.setStatusAsNotReady(config.MLMDProxyReady, err, dspaStatus.SetMLMDProxyStatus)
				return ctrl.Result{}, err
			}
			dspaStatus.SetMLMDProxyStatus(condition)
```

[here](https://github.com/anishasthana/data-science-pipelines-operator/blob/54ea85ecd6100bd556e309ef1cfe1ad136960f4d/controllers/dspipeline_controller.go#L338).